keps/sig-node/20181106-in-place-update-of-pod-resources.md

```diff
@@ -215,13 +270,38 @@ new desired resources (i.e Spec.Containers[i].Resources.Requests) to the sum.
   further action is taken.
   - Kubelet retries the Pod resize at a later time.
 
+<<[UNRESOLVED]>>
+This is close to what I wrote above, but still vague.  What does it mean to
+"reject" a resize?  How do controllers know it has been deferred temporarily
+(will retry) or premantently (no more retries)? Does this put the pod into a
+new condition that higher-level controls need to know?
+<<[/UNRESOLVED]>>
+
 If multiple Pods need resizing, they are handled sequentially in the order in
 which Pod additions and updates arrive at Kubelet.
 
+<<[UNRESOLVED]>>
+I'd call this over-specced.  Kubelet should have freedom to choose higher-prio
+pods or "easy" cases or whatever.
+<<[/UNRESOLVED]>>
+
 Scheduler may, in parallel, assign a new Pod to the Node because it uses cached
 Pods to compute Node allocable values. If this race condition occurs, Kubelet
 resolves it by rejecting that new Pod if the Node has no room after Pod resize.
 
+<<[UNRESOLVED]>>
+What prevents the scheduler from doing the same thing again and again?  This is
+not specific to this KEP, but we don't have a mechanism to prevent this.  E.g.
+we COULD do something like:
+
+* scheduler proposes a pod to a node in a new "earmark" resource
+* kubelet ACKs the earmark with an expiry timestamp
+* scheduler binds the pod to the node
```

Comment:
Our first design proposal and prototype was on similar lines - I [presented it last year](https://kccnceu19.sched.com/event/MPcJ/restart-free-vertical-scaling-for-kubernetes-pods-vinay-kulkarni-futurewei) in KubeCon Barcelona. 

Derek felt the same - more complicated, and once a pod is bound to a node, resizing should be a apiserver <-> node transaction, scheduler may assist if it can (out of scope for this KEP). The general consensus for going with simpler approach was that such a race condition should be a rare thing.

Reply:
> Kubelet ￼resolves it by rejecting that new Pod

What does such a rejection look like?

Reply:
[Pod.Status.Phase = PodFailed](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L1705).

Reply:
ACK.  Though `phase` is overall frowned upon - we should be representing this in a better way, eventually

Reply:
yes, @dashpole had [explained](https://github.com/kubernetes/enhancements/pull/686#discussion_r264914522) to me very well how it has caused problems by requiring knowledge of state. i believe it is tech-debt we have to live with.